Author: Ray Strode <rstrode@redhat.com>
Description: key-file: ignore leading spaces and blank lines
 leading spaces and blank lines currently cause parse errors.
 This commit fixes the parser to be more lenient.
 .
 https://bugs.freedesktop.org/show_bug.cgi?id=67006

diff -Naurp plymouth.orig/src/libply/ply-key-file.c plymouth/src/libply/ply-key-file.c
--- plymouth.orig/src/libply/ply-key-file.c
+++ plymouth/src/libply/ply-key-file.c
@@ -25,6 +25,7 @@
 #include "ply-key-file.h"
 
 #include <assert.h>
+#include <ctype.h>
 #include <errno.h>
 #include <fcntl.h>
 #include <string.h>
@@ -179,7 +180,12 @@ ply_key_file_load_group (ply_key_file_t
       key = NULL;
       value = NULL;
 
-      first_byte = fgetc (key_file->fp);
+      do
+        {
+          first_byte = fgetc (key_file->fp);
+        }
+      while (isspace (first_byte));
+
       if (first_byte == '#')
         {
           char *line_to_toss;
